pkg/policy/selectorcache.go

```diff
@@ -477,12 +518,15 @@ func (sc *SelectorCache) AddFQDNSelector(user CachedSelectionUser, fqdnSelec api
 			cachedSelections: make(map[identity.NumericIdentity]struct{}),
 		},
 		selector: fqdnSelec,
+		dnsProxy: sc.pop,
 	}
 
 	// Add the initial user
 	newFQDNSel.users[user] = struct{}{}
 	newFQDNSel.updateSelections()
 
+	newFQDNSel.NotifyAdded()
+
```

Comment:
`NotifyAdded` calls `updateSelections()` in the end, so maybe don't need both calls to `updateSelections()`?

Reply:
Might be clearer if `NotifyAdded()` was inlined here?

Reply:
Can you clarify what you mean by "inlined"? I removed the call to `updateSelections` as you specified since it was basically a no-op.

Reply:
After removing `updateSelections()` from here it is now hidden in `NotifyAdded()`; it is now a bit less clear that `updateSelections()` is called before returning (as it must be). Moving the body of `NotifyAdded()` in place of the function call to it here would clarify this. Basically `NotifyAdded` is not just notifying, it is also manipulating `newFQDNSel`, which could be surprising to any new reader of this code.

Reply:
Will inline it directly in upcoming commit!